src/api/post/mutation.ts

```diff
@@ -56,19 +56,26 @@ export const usePostLikeMutation = (queryId: string) => {
   const queryClient = useQueryClient();
 
   return useMutation({
-    mutationKey: PostQueryKey.mutate(queryId),
     mutationFn: () => postPostLike(+queryId),
     onMutate: async () => {
-      const previousPost = queryClient.getQueryData(PostQueryKey.mutate(queryId)) as GetPostDetailResponse;
+      await queryClient.cancelQueries({ queryKey: PostQueryKey.detail(+queryId) });
 
-      const newLikeCount = previousPost.isLiked ? previousPost.likeCount - 1 : previousPost.likeCount + 1;
+      const previousPost = queryClient.getQueryData(PostQueryKey.detail(+queryId)) as GetPostDetailResponse;
 
-      const data = produce(previousPost, (draft: GetPostDetailResponse) => {
-        draft.isLiked = !previousPost.isLiked;
-        draft.likeCount = newLikeCount;
+      queryClient.setQueryData(PostQueryKey.detail(+queryId), (oldData: GetPostDetailResponse | undefined) => {
+        if (!oldData) return;
```

Comment:
```suggestion
if (!previousPost) {
  return {previousPost: null};    
}  queryClient.setQueryData(PostQueryKey.detail(+queryId), (oldData: GetPostDetailResponse | undefined) => {
```

1. 현재 데이터가 없다면 setQueryData를 실행할 필요가 없을 것 같아요! early return 시켜주는 게 어떨까요?
2. 없을 때 undefined를 return하는 것보다 명시적으로 null을 리턴시켜주는게 좋아보입니다!

Reply:
1. 데이터가 없는 경우에 아예 실행하지 않는 로직이 더 깔끔할 것 같네요! 수정하겠습니다.
2. `undefined`보다는 말씀해주신 것처럼 `null`이 더 의미상 맞을 것 같아요! 감사합니다

Reply:
굳이 `oldData`를 안에서 체크를 할 필요없이 `previousPost`로 early return 하고 안에서 `oldData`의 `undefined`를 없앴습니다.

괜찮은 방향일까요??

```tsx
if (!previousPost) {
    return { previousPost: null };
  }

  queryClient.setQueryData(PostQueryKey.detail(+queryId), (oldData: GetPostDetailResponse) => {
    return produce(oldData, draft => {
      draft.isLiked = !oldData.isLiked;
      draft.likeCount = oldData.isLiked ? oldData.likeCount - 1 : oldData.likeCount + 1;
    });
  });
```

[99109ce](https://github.com/sopt-makers/sopt-crew-frontend/pull/1186/commits/99109ce77c8b062a77201b61c97f10119ab5e688)